docs/src/main/sphinx/release/release-354.md

```diff
@@ -0,0 +1,56 @@
+# Release 354 (XXXXXXXX Mar 2021)
+
+## General
+
+* Preferred write partitioning is now triggered automatically when the number
+  of written partitions exceeds or is equal to
+  ``preferred-write-partitioning-min-number-of-partitions``. The number of written
+  partitions is estimated from stats, therefore CTAS or insert source tables needs to have
+  statistics collected. To forcefully enable preferred partitioning,
+  ``preferred-write-partitioning-min-number-of-partitions`` needs to be set to
+  ``0`` or ``1``. To fully disable preferred write partitioning, set
+  ``use-preferred-write-partitioning`` to ``false``.
+  ({issue}`6920`)
+* Improve performance of LEFT, RIGHT and FULL JOIN when one side of the join is known to produce a single row. ({issue}`7090`)
```

Comment:
nit: use backticks when mentioning the join types?

Reply:
Isn't there additional condition that the single-row relation has no outputs?

For example, will a JOIN with `count(*)` aggregation over a table (known to produce single row), be improved?

cc @kasiafi @martint

Reply:
It does, but I thought it was too much detail.